src/common/utils/UnhandledExceptionHandlerX64.h

```diff
@@ -0,0 +1,238 @@
+        return "EXCEPTION_ILLEGAL_INSTRUCTION";
+    case EXCEPTION_IN_PAGE_ERROR:
+        return "EXCEPTION_IN_PAGE_ERROR";
+    case EXCEPTION_INT_DIVIDE_BY_ZERO:
+        return "EXCEPTION_INT_DIVIDE_BY_ZERO";
+    case EXCEPTION_INT_OVERFLOW:
+        return "EXCEPTION_INT_OVERFLOW";
+    case EXCEPTION_INVALID_DISPOSITION:
+        return "EXCEPTION_INVALID_DISPOSITION";
+    case EXCEPTION_NONCONTINUABLE_EXCEPTION:
+        return "EXCEPTION_NONCONTINUABLE_EXCEPTION";
+    case EXCEPTION_PRIV_INSTRUCTION:
+        return "EXCEPTION_PRIV_INSTRUCTION";
+    case EXCEPTION_SINGLE_STEP:
+        return "EXCEPTION_SINGLE_STEP";
+    case EXCEPTION_STACK_OVERFLOW:
+        return "EXCEPTION_STACK_OVERFLOW";
+    default:
+        return "UNKNOWN EXCEPTION";
+    }
+}
+
+/* Returns the index of the last backslash in the file path */
+inline int GetFilenameStart(CHAR* path)
+{
+    int pos = 0;
+    int found = 0;
+    if (path != NULL)
+    {
+        while (path[pos] != '\0' && pos < MAX_PATH)
+        {
+            if (path[pos] == '\\')
+            {
+                found = pos + 1;
+            }
+            ++pos;
+        }
+    }
+
+    return found;
+}
+
+inline void LogStackTrace()
+{
+    BOOL result;
+    HANDLE thread;
+    HANDLE process;
+    CONTEXT context;
+    STACKFRAME64 stack;
+    ULONG frame;
+    DWORD64 dw64Displacement;
+    DWORD dwDisplacement;
+
+    memset(&stack, 0, sizeof(STACKFRAME64));
+    memset(pSymbol, '\0', sizeof(*pSymbol) + MAX_PATH);
+    memset(&modulePath[0], '\0', sizeof(modulePath));
+    line.LineNumber = 0;
+
+    try
+    {
+        RtlCaptureContext(&context);
+    }
+    catch (...)
+    {
+        Logger::error(L"Failed to capture context. {}", get_last_error_or_default(GetLastError()));
+        return;
+    }
+    
+    process = GetCurrentProcess();
+    thread = GetCurrentThread();
+    dw64Displacement = 0;
+    stack.AddrPC.Offset = context.Rip;
+    stack.AddrPC.Mode = AddrModeFlat;
+    stack.AddrStack.Offset = context.Rsp;
+    stack.AddrStack.Mode = AddrModeFlat;
+    stack.AddrFrame.Offset = context.Rbp;
+    stack.AddrFrame.Mode = AddrModeFlat;
+
+    std::stringstream ss;
+    for (frame = 0;; frame++)
+    {
+        result = StackWalk64(
+            IMAGE_FILE_MACHINE_AMD64,
+            process,
+            thread,
+            &stack,
+            &context,
+            NULL,
+            SymFunctionTableAccess64,
+            SymGetModuleBase64,
+            NULL);
+
+        if (!result)
+        {
+            break;
+        }
+
+        pSymbol->MaxNameLength = MAX_PATH;
+        pSymbol->SizeOfStruct = sizeof(IMAGEHLP_SYMBOL64);
+
+        if (!SymGetSymFromAddr64(process, stack.AddrPC.Offset, &dw64Displacement, pSymbol))
+        {
+            Logger::error(L"Failed to get a symbol. {}", get_last_error_or_default(GetLastError()));
+        }
+
+        line.LineNumber = 0;
+        SymGetLineFromAddr64(process, stack.AddrPC.Offset, &dwDisplacement, &line);
```

Comment:
Check the `SymGetLineFromAddr64` return value

Reply:
After discussing this, it turned out that when `SymGetLineFromAddr64` fails we simply log `0` has line number.